### PR TITLE
Merge about half of changes from dev

### PR DESCRIPTION
**Pull request by jyn514**

### CODE REVIEW

**Comment by jyn514** [comment]
highlights:
- add rules for downloading html to makefile
- delete html files from git
- add useful links to readme
- add `requirements.py`
- fix many, many bugs
- add an index page (currently unused)

todo before merging:
- get rid of flask
- pull in parsing and schema changes from `dev`; the current one is horribly fragile (it's dependent on how the data was pickled (!!))
- pull in CSV based parsing from `dev` (see https://github.com/jyn514/GradeForge/issues/15)
- make `dump` part of `sql.py` like it is in `dev`
- remove the junk in `query_sql.py`

In short, this is not ready for merge.

**Comment by jyn514** [comment]
This is too much to go through for anyone. Since master hasn't been maintained, I'm going to merge this one time without a review. Future requests will still require reviews.
